File: Stack/Valid_Parentheses/1_BruteForce.cpp
#include <string>
#include <algorithm>
using namespace std;

class Solution {
public:
    bool isValid(string s) {
        const string pairs[3] = {"()", "{}", "[]"};
        bool changed = true;
        while (changed) {
            changed = false;
            for (const string &p : pairs) {
                size_t pos = s.find(p);
                if (pos != string::npos) {
                    s.erase(pos, 2);
                    changed = true;
                }
            }
        }
        return s.empty();
    }
};

File: Stack/Valid_Parentheses/1_BruteForce.java
public class Solution {
    public boolean isValid(String s) {
        String[] pairs = {"()", "{}", "[]"};
        boolean changed = true;
        while (changed) {
            changed = false;
            for (String p : pairs) {
                int idx = s.indexOf(p);
                if (idx != -1) {
                    s = s.substring(0, idx) + s.substring(idx + 2);
                    changed = true;
                }
            }
        }
        return s.isEmpty();
    }
}
